### DISCUSSION TITLE
Make a matching generic_edge_bfs to accompany generic_bfs_edge

### DISCUSSION THREAD

**Discussion by Talos878** [networkx/networkx#5270] [Algorithms]
There are two algorithms, edge_bfs and bfs_edge, which do breath first traversals of graphs. The first will show edges connecting to already searched nodes, while the second one does not. (see the documentation in their source code for a more thorough explanation)

There's also a function generic_bfs_edges which can take an arbitrary function for getting that neighbors of a node. However there is no accompanying generic function for edges_bfs. 

I am hoping to quickly implement a generic version of the missing function.

**Comment by dschult** [comment]
This seems like a good idea to me.

I guess it should go in the module `edgebfs.py`.   Add the function name to the file `doc/reference/algorithms/traversal.rst` to connect the new function to the documentation.
:}
Thanks!

**Comment by MridulS** [comment]
@Talos878 I have created an issue to track this https://github.com/networkx/networkx/issues/5295, do tag the issue when you submit a PR, or you can use the issue to ask more questions about it. Thanks!

**Comment by Ku6er** [comment]
Hi @dschult,
bfs_edge calls generic_bfs_edges to run bfs, however in edge_bfs, the code for bfs is written inside this function. Is the change expected to break edge_bfs into two functions?

**Reply by dschult** [reply]
You will be creating a new function for `generic_edge_bfs` right? So, if that new function includes the code that is currently in `edge_bfs` and it is easy to have `edge_bfs` call `generic_edge_bfs`, that sounds like a good way to reduce duplicated code. If that is what you mean by splitting  `edge_bfs` into 2 functions, then "yes" you should probably do that.  :}

**Comment by Talos878** [comment]
Sorry for being MIA, @JatinSharma I have a PR that I will submit now that breaks the function but the way the code was originally written, splitting it into two comes out kinda unclean. I have had some nerve damage problems the last few weeks so I haven't been able to get around to it. 

This is the branch in my forked github: https://github.com/Talos878/networkx/tree/feature-issue5270

I am can make a pull request but I would recommend someone looking over the code before submission for sure, because the generic function call pattern is just slightly different from how the generic functions work in other examples and I think it could be improved.

**Reply by rossbar** [reply]
> I am can make a pull request but I would recommend someone looking over the code before submission for sure, because the generic function call pattern is just slightly different from how the generic functions work in other examples and I think it could be improved.

@Talos878 please feel free to just open a PR if you feel it's in a state where it's ready to be looked at by others. It's *much* easier to review/understand changes when there are diffs + tests etc. which you get automatically when opening a PR!

**Comment by Talos878** [comment]
Whatever you guys need to get access to what I have done so far and improve it, just let me know

**Reply by dschult** [reply]
Yes, please make a PR so we can see the differences, discuss what was changed between the new function and the old, etc. Don't worry about it not being quite ready to merge. That's what a PR is for ... to make it easy to discuss the code before it gets merged. If you are pretty sure it isn't ready to be merged, click on the arrow on the right side of the "Create a Pull Request" button and one option is to create a "draft" pull request. Then we can still look at and comment on the work, but we also know that it isn't ready to be merged yet.
